lib/cynic.dart

```diff
@@ -17,31 +16,50 @@ const Service _googleDns = const Service.forIp('8.8.8.8');
 /// servers as proof of internet connectivity. You will need to implement custom
 /// functionality for an intranet.
 Future<bool> isOnline({Duration timeout}) {
-  return _googleDns.isOnline(timeout: timeout);
+  return Reachable.googleDns.isOnline(timeout: timeout);
 }
 
 /// Returns a future that completes with whether the internet is not reachable.
 ///
 /// See [isOnline] for full documentation.
 Future<bool> isOffline({Duration timeout}) async {
-  return _googleDns.isOffline(timeout: timeout);
+  return Reachable.googleDns.isOffline(timeout: timeout);
 }
 
 /// Platform-agnostic indication of a remote service being online or offline.
-abstract class Service {
+abstract class Reachable {
+
+  static const Reachable googleDns = const Reachable.ip('8.8.8.8');
+
   /// Returns a future that completes if all [services] are online.
-  static Future<bool> allOnline(
-    Iterable<Service> services, {
+  static Future<bool> anyOnline(Iterable<Reachable> services, {
     Duration timeout,
-  }) async =>
-      !(await Future.any(services.map((s) => s.isOffline(timeout: timeout))));
+  }) async {
+    final List<bool> executions = await Future.wait(
```

Comment:
Does it mean if all futures have completed and none is confirmed then complete with false ?
```dart 
Future.wait(futures).then((all) {
  _maybeComplete(false);
});
```
So instead
```dart 
Future.wait(futures).then((all) {
  completer.complete(false);
});
```